### DISCUSSION TITLE
buildLink gone?

### DISCUSSION THREAD

**Discussion by Jarzka** [TanStack/router#1215] [Q&A]
Hi

After trying to upgrade from `0.0.1-beta.185` to `1.16.6` I noticed that `buildLink` function of `useRouter` is gone. We have used it to generate link props for our own `Anchor` component. Like this:

```
const { buildLink } = useRouter();

<Anchor
  {...linkInfoToAnchorProps(
    buildLink({
      to: userRouteId,
      params: {
        language
      },
    })
  )}
>

function linkInfoToAnchorProps(linkInfo: LinkInfo): Partial<AnchorProps> {
  if (linkInfo.type === "external") {
    return {
      href: linkInfo.href,
    };
  } else {
    return {
      href: linkInfo.next.href,
      onFocus: (event) => linkInfo.handleFocus(event),
      onClick: (event) => linkInfo.handleClick(event),
      onMouseEnter: (event) => linkInfo.handleEnter(event),
      onMouseLeave: (event) => linkInfo.handleLeave(event),
      onTouchStart: (event) => linkInfo.handleTouchStart(event),
    };
  }
}
```

I noticed that there is now a built-in `Link` component. However, we would prefer using our own `Anchor` component. It is defined in an external component library, which does and should not know anything about TanStack Router.

What is the current way to build custom links via props instead of using the built-in `Link` component?

EDIT: We have also used the `useLinkProps` hook function for generating link props. It could probably replace `buildLink`, but since it is a hook, it cannot be used inside utility functions which use `buildLink` for generating links with certain parameters. 😞

**Comment by Jarzka** [comment]
Since this function seems to be gone, we went and spent a whole work-week refactoring all our `buildLink` link props to `Link` component :feelsbadman:

**Comment by mrlubos** [comment]
Bumping this. Is there a recommended way to generate final route paths in a typesafe manner?
